clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -7823,7 +7839,8 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
 
   TranslatorArgs.push_back("-o");
   TranslatorArgs.push_back(Output.getFilename());
-  if (getToolChain().getTriple().isSYCLDeviceEnvironment()) {
+  if (getToolChain().getTriple().isSYCLDeviceEnvironment() ||
+      TCArgs.hasArg(options::OPT_fsycl_device_only)) {
```

Comment:
Would the phase graph run up to the translator with `-fsycl-device-only`, though?

Reply:
Yes - this can occur with the usage of `-fno-sycl-use-bitcode`.  `-fsycl-use-bitcode` is the default.

Reply:
Could this be tested alongside the regular behavior?

Reply:
This is already tested here https://github.com/intel/llvm/blob/692f758c5c5994b93f30953bba700231a6c4f748/clang/test/Driver/sycl.c#L48